src/main/java/com/wultra/app/enrollmentserver/controller/api/PushRegistrationController.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.app.enrollmentserver.controller.api;
+
+import com.wultra.app.enrollmentserver.model.request.PushRegisterRequest;
+import com.wultra.app.enrollmentserver.errorhandling.InvalidRequestObjectException;
+import com.wultra.app.enrollmentserver.errorhandling.PushRegistrationFailedException;
+import com.wultra.app.enrollmentserver.model.validator.PushRegisterRequestValidator;
+import io.getlime.core.rest.model.base.request.ObjectRequest;
+import io.getlime.core.rest.model.base.response.Response;
+import io.getlime.push.client.MobilePlatform;
+import io.getlime.push.client.PushServerClient;
+import io.getlime.push.client.PushServerClientException;
+import io.getlime.security.powerauth.crypto.lib.enums.PowerAuthSignatureTypes;
+import io.getlime.security.powerauth.rest.api.base.authentication.PowerAuthApiAuthentication;
+import io.getlime.security.powerauth.rest.api.base.exception.PowerAuthAuthenticationException;
+import io.getlime.security.powerauth.rest.api.spring.annotation.PowerAuthToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+/**
+ * Controller with services related to Push Server registration.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+@Controller
+@RequestMapping(value = "api/push")
+public class PushRegistrationController {
+
+    private static final Logger logger = LoggerFactory.getLogger(PushRegistrationController.class);
+
+    private PushServerClient client;
+
+    @Autowired
+    public PushRegistrationController(PushServerClient client) {
+        this.client = client;
+    }
+
+    @RequestMapping(value = "device/register", method = RequestMethod.POST)
+    @PowerAuthToken(signatureType = {
+            PowerAuthSignatureTypes.POSSESSION,
+            PowerAuthSignatureTypes.POSSESSION_BIOMETRY,
+            PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE,
+            PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE_BIOMETRY
+    })
```

Comment:
Do we really want to allow all signature types? I am not sure that for instance `POSSESSION_KNOWLEDGE_BIOMETRY` is useful.